app/models/lion.rb

```diff
@@ -28,35 +36,19 @@ def self.create_from_image_set(image_set, name)
   end
 
   def self.search(search_params)
-    age_params = search_params.extract!(:dob_range_start, :dob_range_end)
-
-    unless age_params.empty?
-      dob_start = DateTime.parse(age_params[:dob_range_start])
-      dob_end = DateTime.parse(age_params[:dob_range_end])
-      dob_hash = { date_of_birth: (dob_start..dob_end) }
-      search_params.merge!(dob_hash)
-    end
-
-    params = {}
+    lion_params = {}
     LION_SEARCH_PARAMS.each do |key|
-      params[key] = search_params.delete key if search_params.has_key? key
+      lion_params[key] = search_params.delete key if search_params.has_key? key
     end
 
     unless search_params.empty?
-      params[:image_sets] = search_params
+      image_sets = ImageSet.search(search_params)
+      lion_params[:primary_image_set] = image_sets.collect { |img| img.id }
```

Comment:
`img` for `image_set` in the block at the end is a confusing name. You can skip it with a symbol proc: `image_sets.collect &:id`

I'm not sure what the performance characteristics of an `in` query are — this would generate an `image_set_id IN [...]` query, right? Does it perform well if the list of matching image set ids is very large?